src/pages/RequestCallPage.js

```diff
@@ -0,0 +1,176 @@
+import CONST from '../CONST';
+import Growl from '../libs/Growl';
+import {requestConciergeDMCall} from '../libs/actions/Inbox';
+import {fetchOrCreateChatReport} from '../libs/actions/Report';
+
+const propTypes = {
+    ...withLocalizePropTypes,
+
+    /** The personal details of the person who is logged in */
+    myPersonalDetails: PropTypes.shape({
+        /** Display name of the current user from their personal details */
+        displayName: PropTypes.string,
+
+        /** Primary login of the current user */
+        login: PropTypes.string,
+    }),
+
+    /** Current user session */
+    session: PropTypes.shape({
+        email: PropTypes.string.isRequired,
+    }).isRequired,
+};
+const defaultProps = {
+    myPersonalDetails: {},
+};
+
+class RequestCallPage extends React.Component {
+    constructor(props) {
+        super(props);
+
+        // The displayName defaults to the user's login if they haven't set a first and last name,
+        // which we can't use to prefill the input fields
+        const [firstName, lastName] = props.myPersonalDetails.displayName !== props.myPersonalDetails.login
+            ? props.myPersonalDetails.displayName.split(' ')
+            : [];
+        this.state = {
+            firstName: firstName ?? '',
+            lastName: lastName ?? '',
+            phoneNumber: this.getPhoneNumber(props.user.loginList) ?? '',
+            isLoading: false,
+        };
+
+        this.onSubmit = this.onSubmit.bind(this);
+        this.getPhoneNumber = this.getPhoneNumber.bind(this);
+    }
+
+    onSubmit() {
+        this.setState({isLoading: true});
+        if (!this.state.firstName.length || !this.state.lastName.length) {
+            Growl.show(this.props.translate('requestCallPage.growlMessageEmptyName'), CONST.GROWL.ERROR);
+            this.setState({isLoading: false});
+            return;
+        }
+
+        requestConciergeDMCall('', this.state.firstName, this.state.lastName, this.state.phoneNumber)
+            .then((result) => {
+                this.setState({isLoading: false});
+                if (result.jsonCode === 200) {
+                    Growl.show(this.props.translate('requestCallPage.growlMessageOnSave'), CONST.GROWL.SUCCESS);
+                    fetchOrCreateChatReport([this.props.session.email, CONST.EMAIL.CONCIERGE], true);
+                    return;
+                }
+
+                // Phone number validation is handled by the API
+                Growl.show(result.message, CONST.GROWL.ERROR, 3000);
+            });
+    }
+
+    /**
+     * Gets the user's phone number from their secondary login.
+     * Returns null if it doesn't exist.
+     * @param {Array<Object>} loginList
+     *
+     * @returns {String|null}
+     */
+    getPhoneNumber(loginList) {
+        const secondaryLogin = _.find(loginList, login => Str.isSMSLogin(login.partnerUserID));
+        return secondaryLogin ? Str.removeSMSDomain(secondaryLogin.partnerUserID) : null;
```

Comment:
Maybe something to consolidate with Joe's code ?

Reply:
Joe's code assumes that the user is an SMS user, so it isn't entirely possible to reuse what he has.